src/interactive-auth.js

```diff
@@ -318,14 +318,18 @@ InteractiveAuth.prototype = {
      */
     _doRequest: async function(auth, background) {
         try {
+            if (Object.keys(auth).length === 0 &&
+                auth.constructor === Object) auth = null;
```

Comment:
I don't think we need to modify `auth` here - we should be supplying the function with what it needs.

Reply:
I think you are right. This seems to be a bad workaround that I made. I should do it properly as I probably now broke things without realising.